File: apps/client/src/app/dto/review.ts
import { IStatComparisonDto } from "./common";
import { IScoreDistributionDto } from "./statistic";

export interface IReviewDto {
    averageScore: IStatComparisonDto<number>;
    scoreDistribution: IScoreDistributionDto[];
    filterRangeDays: number;
    activities: IPaginatedResultDto<IActivityReviewDto>;
}

export interface IActivityReviewDto {
    activityId: string;
    activityName: string;
    averageScore: IStatComparisonDto<number>;
    totalReviews: number;
}

export interface IPaginatedResultDto<T> {
    items: T[];
    totalCount: number;
    page: number;
    pageSize: number;
    hasMore: boolean;
}

File: apps/api/src/main/kotlin/com/wewiins/saas_api/interfaces/Review.kt
package com.wewiins.saas_api.interfaces

data class Review(
    val averageScore: ComparisonStat<Double>,
    val scoreDistribution: List<ScoreDistribution>,
    val filterRangeDays: Int,
    val activities: PaginatedResult<ActivityReview>
)

data class ActivityReview(
    val activityId: String,
    val activityName: String,
    val averageScore: ComparisonStat<Double>,
    val totalReviews: Int
)

data class PaginatedResult<T>(
    val items: List<T>,
    val totalCount: Int,
    val page: Int,
    val pageSize: Int,
    val hasMore: Boolean
)